samples: rename template to sid_end_device (#494)
nrfconnect/sdk-sidewalk: doc/samples/sid_end_device.rst

Reviewer: This graph look strange.
![image](https://github.com/nrfconnect/sdk-sidewalk/assets/85483674/502a3823-4fb8-4e68-997f-4214425c890f)
Are you sure this is correct one? It does not describe much.

Author: It describes a thread has tow stages, starts in sidewalk, and the same dfu event is used to enter end exit dfu state. The source file has over 300 lines of code, but the main idea is presented on this small diagram.
If you by mean colors and layout, maybe it is possible to generate something better looking but I'm not sure

.. _sidewalk_end_device:

Sidewalk End Device
###################

This sample demonstrates a end device for the Sidewalk End Device application.
It supports Bluetooth LE, LoRa, and FSK link modes.

> External QSPI Flash is used for firmware updates.

Reviewer: ```suggestion

```

Author: The sample is using qspi flash for device updates, why should we remove this line?

Reviewer: In my opinion, this information is already in table in sample_list but it can stay

> 

Author: I'm not sure, I think we link to the migration guide before, there is no need to repeat this image here.
Same thing with zephyr state machine - it is mentioned in migration guide. Also we don't mention another sdk api we are using in samples like zephyr logs, dk buttons and leds, or even state notifier and json printer. I think developers can see such dependencies looking in config files or #Incldudes

> 

Reviewer: yes, but in the future we will remove migration guide and the info there will be lost.
Up to you

* ``CONFIG_SIDEWALK_DFU`` -- Enables the nRF Connect SDK bootloader and DFU service over Bluetooth LE.

* ``CONFIG_SID_END_DEVICE`` -- Switch between different application types.

>    * ``CONFIG_SID_END_DEVICE_HELLO`` -- Enables simple Sidewalk hello world application. This is default option. See :ref:`sidewalk_hello` for more information.
>    * ``CONFIG_SID_END_DEVICE_SENSOR_MONITORING`` -- Enables Sidewalk Sensor monitoring application. See :ref:`sidewalk_demo` for more information.
>    * ``CONFIG_SID_END_DEVICE_DUT`` -- Enables Sidewalk device under test application. See :ref:`sidewalk_dut` for more information.

Reviewer 2: ```suggestion
   * ``CONFIG_SID_END_DEVICE_HELLO`` -- Enables the Hello Sidewalk application. 
     This is the default option. 
     For more details, see the :ref:`sidewalk_hello` page.
   * ``CONFIG_SID_END_DEVICE_SENSOR_MONITORING`` -- Enables the Sidewalk Sensor monitoring application. 
     For more details, see the :ref:`sidewalk_demo` page.
   * ``CONFIG_SID_END_DEVICE_DUT`` -- Enables the Sidewalk device under test application. 
     For more details, see the :ref:`sidewalk_dut` page.
```

> * ``CONFIG_SID_END_DEVICE_CLI`` -- Enables Sidewalk CLI. To see the list of available commands, flash sample and type ``sid help``.

Reviewer 2: ```suggestion
* ``CONFIG_SID_END_DEVICE_CLI`` -- Enables Sidewalk CLI. 
  To see the list of available commands, flash the sample and type ``sid help``.
```

> * ``CONFIG_SID_END_DEVICE_AUTO_START`` -- Enable automatic Sidewalk initialization and start.

Reviewer 2: ```suggestion
* ``CONFIG_SID_END_DEVICE_AUTO_START`` -- Enables an automatic Sidewalk initialization and start.
```

* ``CONFIG_SID_END_DEVICE_AUTO_CONN_REQ`` -- Enable automatic connection request before message send. The Bluetooth LE connection request action is performed automatically if needed.

> You can build the ``sensor_monitoring`` end device application, with Bluetooth LE only libraries for ``build_target`` by running the following command in the project directory:

Reviewer 2: ```suggestion
You can build the ``sensor_monitoring`` end device application with Bluetooth LE only libraries by running the following command in the project directory:
```

> 

Reviewer: * ``CONFIG_SIDEWALK_SUBGHZ_SUPPORT`` -- Enables use of Sidewalk libraries with Bluetooth LE support only.

I think it should be:
* ``CONFIG_SIDEWALK_SUBGHZ_SUPPORT`` -- Enables use of Sidewalk libraries with Bluetooth LE, LoRa and FSK support. Disabling it causes to use of Sidewalk libraries with Bluetooth LE support only.

      Sidewalk --> FirmwareUpdate : event_dfu
      FirmwareUpdate --> Sidewalk : event_dfu

.. _sidewalk_end_device_building_and_running:

> Building and Running

Reviewer: This section is empty.
Shall it stays or shall it goes away?
